lib/Mojolicious/Plugin/OpenAPI/Guides/OpenAPI3.pod

```diff
@@ -0,0 +1,392 @@
+
+  "x-mojo-to": ["pet#list", {"foo": 123}]
+  $route->to("pet#list", {foo => 123});
+
+=item * security and securitySchemes
+
+The securityScheme is added under components, where on way is to have the client
+place an apiKey in the header of the request
+
+  {
+    ...
+    "components": {
+      "securitySchemes": {
+        "apiKey": {
+          "name": "X-Api-Key",
+          "in": "header",
+          "type": "apiKey"
+        }
+      }
+    }
+  }
+
+Its then referenced under the path object as security like this
+
+  {
+    ...
+    "paths": {
+      "/pets": {
+        "get": {
+          "operationId": "getPets",
+          ...
+          "security": [
+            {
+              "apiKey": []
+            }
+          ]
+        }
+      }
+    }
+  }
+
+You can then utilize security, by adding security callback when loading
+the plugin
+
+  $self->plugin(
+    OpenAPI => {
+      spec     => $self->static->file("openapi.json")->path,
+      schema   => 'v3',
+      security => {
+        apiKey => sub ($c, $definition, $scopes, $cb) {
+          if (my $key = $c->tx->req->content->headers->to_hash->{'X-Api-Key'}) {
+            if (got_valid_api_key()) {
+              return $c->$cb();
+            }
+            else {
+              return $c->$cb('Api Key not valid');
+            }
+          }
+          else {
+            return $c->$cb('Api Key header not present');
+          }
+        }
+      }
+    }
+  );
+
+=back
+
+=head2 Application
+
+  package Myapp;
+  use Mojo::Base "Mojolicious";
+
+  sub startup {
+    my $app = shift;
+    $app->plugin("OpenAPI" => {url => $app->home->rel_file("myapi.json"), schema => 'v3'});
+  }
+
+  1;
+
+The first thing in your code that you need to do is to load this plugin and the
+L</Specification>. See L<Mojolicious::Plugin::OpenAPI/register> for information
+about what the plugin config can be.
+
+See also L<Mojolicious::Plugin::OpenAPI/SYNOPSIS> for example
+L<Mojolicious::Lite> application.
+
+=head2 Controller
+
+  package Myapp::Controller::Pet;
+  use Mojo::Base "Mojolicious::Controller";
+
+  sub list {
+
+    # Do not continue on invalid input and render a default 400
+    # error document.
+    my $c = shift->openapi->valid_input or return;
+
+    # You might want to introspect the specification for the current route
+    my $spec = $c->openapi->spec;
+    unless ($spec->{'x-opening-hour'} == (localtime)[2]) {
+      return $c->render(openapi => [], status => 498);
+    }
+
+    # $c->openapi->valid_input copies valid data to validation object,
+    # and the normal Mojolicious api works as well.
+    my $input = $c->validation->output;
+    my $age   = $c->param("age"); # same as $input->{age}
+    my $body  = $c->req->json;    # same as $input->{body}
+
+    # $output will be validated by the OpenAPI spec before rendered
+    my $output = {pets => [{name => "kit-e-cat"}]};
+    $c->render(openapi => $output);
+  }
+
+  1;
```

Comment:
Do we need the "1;" at the end?

Reply:
not sure, it just taken from your Tutorial.pod :) but just trying to remove it, will produce an "return a true value at" error at execution